sleap/gui/commands.py

```diff
@@ -1488,30 +1489,106 @@ def do_action(context: CommandContext, params: dict):
+
+        def _trim_labeled_frames(lfs_to_remove):
+            """Trim labels past new video length."""
+            for lf in lfs_to_remove:
+                context.labels.remove_frame(lf)
+
+        def _reset_video_backend(video, filename, grayscale):
+            """Reset video back to original if operation aborted."""
+            video.backend.reset(
+                filename=filename,
+                grayscale=grayscale,
+            )
+
+        import_list = params["import_list"]
+
+        for import_item, video in import_list:
+            import_params = import_item["params"]
+            old_path = video.backend.filename
+            old_grayscale = video.backend.grayscale
+            new_path = import_params["filename"]
+
+            # TODO: Will need to create a new backend if import has different extension.
+            # Currently reset is only functional for MediaVideo backend.
+            # FIXME: Ideally we would only mess with the backend after getting approval
+            # ...but, use this to read in the number of frames. Could use cv2 here....
+            if new_path != video.backend.filename:
+                _reset_video_backend(video, new_path, import_params)
+
+            # Remove frames in video past last frame index
+            last_vid_frame = video.last_frame_idx
+            lfs: List[LabeledFrame] = list(context.labels.get(video))
+            if lfs is not None:
+                lfs.sort(key=lambda lf: lf.frame_idx)
+                last_lf_frame = lfs[-1].frame_idx
+                lfs = [lf for lf in lfs if lf.frame_idx > last_vid_frame]
+
+                # Warn users that labels will be removed if proceed
+                if last_lf_frame > last_vid_frame:
+                    message = (
+                        "<p><strong>Warning:</strong> The replacement video is shorter "
+                        f"than the frame index of {len(lfs)} labeled frames.</p>"
+                        f"<p><em>Current video</em>: <b>{Path(old_path).name}</b> "
+                        f"(last label at frame {last_lf_frame})<br>"
+                        f"<em>Replacement video</em>: <b>{Path(video.filename).name}"
+                        f"</b> ({last_vid_frame} frames)</p>"
+                        f"<p>Replace video (and remove {len(lfs)} labeled frames past "
+                        f"frame {last_vid_frame})?</p>"
+                    )
+                    query = QueryDialog(
+                        "Replace Video: Truncate Labeled Frames", message, context.app
+                    )
+                    query.accepted.connect(lambda: _trim_labeled_frames(lfs))
+                    # FIXME: Ideally we do not mess with backend before approval
+                    query.rejected.connect(
+                        lambda: _reset_video_backend(video, old_path, old_grayscale)
+                    )
+                    query.exec_()
```

Comment:
This bit of code will pop up for EVERY video that has labeled frames past the replacement video length - is it safe to assume that there will not be many cases where the above is true? How often do our users replace videos? How many videos do they replace at once? Otherwise, we will need to create a pop-up box similar to the `ImportItemWidget` which shows all problem videos in a single dialog.

Reply:
The warning is now built into the `ImportItemWidget` that pops up to create our list of files to import.